script/download-file/customize.py

```diff
@@ -195,19 +219,26 @@ def preprocess(i):
     if env.get('CM_DOWNLOAD_CHECKSUM_FILE', '') != '':
         env['CM_DOWNLOAD_CHECKSUM_CMD'] = f"cd {q}{filepath}{q} {xsep}  md5sum -c {x_c} {x}{q}{env['CM_DOWNLOAD_CHECKSUM_FILE']}{q}"
     elif env.get('CM_DOWNLOAD_CHECKSUM', '') != '':
-        env['CM_DOWNLOAD_CHECKSUM_CMD'] = "echo {} {}{}{}{} | md5sum {} -c -".format(env.get('CM_DOWNLOAD_CHECKSUM'), x, q, env['CM_DOWNLOAD_FILENAME'], q, x_c)
+        if os_info['platform'] == 'windows':
+            env['CM_DOWNLOAD_CHECKSUM_CMD'] = "echo {} {}{} | md5sum {} -c -".format(env.get('CM_DOWNLOAD_CHECKSUM'), x, escape_special_chars(env['CM_DOWNLOAD_FILENAME']), x_c)
+        else:
+            env['CM_DOWNLOAD_CHECKSUM_CMD'] = "echo {} {}{}{}{} | md5sum {} -c -".format(env.get('CM_DOWNLOAD_CHECKSUM'), x, q, env['CM_DOWNLOAD_FILENAME'], q, x_c)
         for i in range(1,5):
             if env.get('CM_DOWNLOAD_CHECKSUM'+str(i),'') == '':
                 break
-            env['CM_DOWNLOAD_CHECKSUM_CMD'] += " || echo {} {}{}{}{} | md5sum {} -c -".format(env.get('CM_DOWNLOAD_CHECKSUM'+str(i)), x, q, env['CM_DOWNLOAD_FILENAME'], q, x_c)
+            if os_info['platform'] == 'windows':
+                env['CM_DOWNLOAD_CHECKSUM_CMD'] += " || echo {} {}{} | md5sum {} -c -".format(env.get('CM_DOWNLOAD_CHECKSUM'+str(i)), x, escape_special_chars(env['CM_DOWNLOAD_FILENAME']), x_c)
+            else:
+                env['CM_DOWNLOAD_CHECKSUM_CMD'] += " || echo {} {}{}{}{} | md5sum {} -c -".format(env.get('CM_DOWNLOAD_CHECKSUM'+str(i)), x, q, env['CM_DOWNLOAD_FILENAME'].replace("%", "%%"), q, x_c)
+        # print(env['CM_DOWNLOAD_CHECKSUM_CMD'])
     else:
         env['CM_DOWNLOAD_CHECKSUM_CMD'] = ""
 
     if not pre_clean:
         env['CM_PRE_DOWNLOAD_CMD'] = ''
 
     if os_info['platform'] == 'windows' and env.get('CM_DOWNLOAD_CMD', '') != '':
-        env['CM_DOWNLOAD_CMD'] =  env['CM_DOWNLOAD_CMD'].replace('&', '^&').replace('|', '^|').replace('(', '^(').replace(')', '^)')
+        env['CM_DOWNLOAD_CMD'] = escape_special_chars(env['CM_DOWNLOAD_CMD'], tool)
```

Comment:
Hi @anandhu-eng as I understand, we need to escape special characters when we use '"' and we don't need to escape them when we don't use '"'. So, is there a reason or advantage of using '"' like when we have space in the file paths?

Reply:
Hi @arjunsuresh , when tried  a simple command to change directory to a folder which has space in its name, `powershell` requied the use of quotes for the entire path but the same functionality worked without quotes in `command prompt`.

Reply:
Oh okay.. Then I think the current code is fine. Perhaps you can add a comment saying for handling space in paths on Dos.

Reply:
Also @anandhu-eng, the mlperf workflow is working fine on Windows via Github action even without this change. So I believe the problem is specific to Dos shell as GH action uses Power Shell.